ruby/private/bundle/create_bundle_build_file.rb

```diff
@@ -161,35 +170,31 @@ def initialize(workspace_name:,
   def generate!
     # when we append to a string many times, using StringIO is more efficient.
     template_out = StringIO.new
-
-    # In Bazel we want to use __FILE__ because __dir__points to the actual sources, and we are
-    # using symlinks here.
-    #
-    # rubocop:disable Style/ExpandPathArguments
-    bin_folder = File.expand_path('../bin', __FILE__)
-    binaries   = Dir.glob("#{bin_folder}/*").map do |binary|
-      'bin/' + File.basename(binary) if File.executable?(binary)
-    end
-    # rubocop:enable Style/ExpandPathArguments
-
-    template_out.puts TEMPLATE
+    template_out.puts BUILD_HEADER
                         .gsub('{workspace_name}', workspace_name)
                         .gsub('{repo_name}', repo_name)
                         .gsub('{ruby_version}', ruby_version)
-                        .gsub('{binaries}', binaries.to_s)
                         .gsub('{bundler_setup}', bundler_setup_require)
 
     # strip bundler version so we can process this file
     remove_bundler_version!
+
     # Append to the end specific gem libraries and dependencies
     bundle        = Bundler::LockfileParser.new(Bundler.read_file(gemfile_lock))
     gem_lib_paths = []
-    bundle.specs.each { |spec| register_gem(spec, template_out, gem_lib_paths) }
+    gems_binaries = {} # gem-name => [ gem's binaries ], ...
+    gems          = bundle.specs.map(&:name)
+
+    bundle.specs.each { |spec| register_gem(spec, template_out, gem_lib_paths, gems_binaries) }
 
     template_out.puts ALL_GEMS
-                        .gsub('{gems_lib_files}', gem_lib_paths.map { |p| "#{p}/**/*.rb" }.to_s)
+                        .gsub('{gems_lib_files}', to_flat_string(gem_lib_paths.map { |p| "#{p}/**/*" }))
+                        .gsub('{gems_binaries}', gems_binaries.values.flatten.to_s)
```

Comment:
This is incorrect, doesnt match the string in the template

Reply:
Good catch!